compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -765,18 +784,27 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
         // `covered == Empty` may happen for primitive types with auto-conversion
         // see tests/patmat/reader.scala  tests/patmat/byte.scala
-        if (covered == Empty) covered = curr
+        if (covered == Empty && !isNullLit(pat)) covered = curr
 
         if (isSubspace(covered, prevs)) {
           ctx.warning(MatchCaseUnreachable(), pat.sourcePos)
         }
 
         // if last case is `_` and only matches `null`, produce a warning
-        if (i == cases.length - 1 && !isNull(pat) ) {
-          simplify(minus(covered, prevs)) match {
-            case Typ(`nullType`, _) =>
-              ctx.warning(MatchCaseOnlyNullWarning(), pat.sourcePos)
-            case _ =>
+        if (i == cases.length - 1 && !isNullLit(pat) ) {
+          val simpl = simplify(minus(covered, prevs))
+          if (ctx.settings.YexplicitNulls.value) {
+            simpl match {
+              case space if isNullSpace(space) =>
+                ctx.warning(MatchCaseOnlyNullWarning(), pat.sourcePos)
+              case _ =>
+            }
```

Comment:
If `ctx. YexplicitNulls`, we can skip the check for `_` matching `null` -- most of such cases will get unreachable warnings.

Reply:
Done. Added a comment explaining why the check isn't needed.